services/graph/pkg/identity/ldap.go

```diff
@@ -131,6 +171,10 @@ func (i *LDAP) CreateUser(ctx context.Context, user libregraph.User) (*libregrap
 		return nil, ErrReadOnly
 	}
 
+	if user.AccountEnabled != nil && i.disableUserMechanism == DisableMechanismNone {
+		return nil, errors.New("accountEnabled option not compatible with backend disable user mechanism")
```

Comment:
Is this really needed, if `disabledUserMechanism` is none, isn't it essentially just "all users are enabled", so this flag would be invisible at worst.

I can also see someone adding all the users with this flag set first, before turning on the disable mechanism.

Or is this here because if none, we don't know how to disable a user?

And if that's the case, why not just ignore the flag so that things using the API don't have to know how this is set just to add a user without checking for this error.

Reply:
Hm, there are good reasons for both options. 

If we don't do the check here. We need at least to make sure to not set the attribute in LDAP. Also the value should be absent from the user that is returned from CreateUser response.

But I am also fine with keeping the current behavior.

Reply:
Yeah if set to `none`, I'd personally choose to just ignore it in adding it to LDAP, and always setting it to `true` in the responses, as I see `none` as "everything is enabled" and having the API change, even if slightly, because of this setting would make code using the API a bit more complex where it doesn't really have to IMHO.